modular_skyrat/modules/assault_operatives/code/equipment_items/guns.dm

```diff
@@ -0,0 +1,227 @@
+// La Pistola
+
+/obj/item/gun/ballistic/automatic/pistol/clandestine/assault_ops
+	name = "\improper IGE-040 pistol"
+	desc = "A pistol chambered in 10mm magnum and painted in an ominous matte black. Strangely, the gun also seems to lack any form of manufacturer markings."
+	company_flag = COMPANY_REMOVED
+
+/obj/item/gun/energy/e_gun/advtaser/assault_ops
+	w_class = WEIGHT_CLASS_NORMAL
+	company_flag = COMPANY_REMOVED
+
+// Rifle
+
+/obj/item/gun/ballistic/automatic/assault_ops_rifle
+	name = "\improper IGE-110 rifle"
+	desc = "A bullpup rifle chambered in 5.6x40mm and painted in an ominous matte black. Strangely, the gun also seems to lack any form of manufacturer markings."
+	icon_state = "ige_assault"
+	icon = 'modular_skyrat/modules/assault_operatives/icons/guns/guns.dmi'
+	inhand_icon_state = "ige_assault"
+	righthand_file = 'modular_skyrat/modules/assault_operatives/icons/guns/guns_righthand.dmi'
+	lefthand_file = 'modular_skyrat/modules/assault_operatives/icons/guns/guns_lefthand.dmi'
+	worn_icon_state = "ige_assault"
+	worn_icon = 'modular_skyrat/modules/assault_operatives/icons/guns/guns_worn.dmi'
+	base_pixel_x = -8
+	w_class = WEIGHT_CLASS_BULKY
+	slot_flags = ITEM_SLOT_BACK | ITEM_SLOT_OCLOTHING
+	mag_type = /obj/item/ammo_box/magazine/multi_sprite/assault_ops_rifle
+	fire_sound = 'modular_skyrat/modules/sec_haul/sound/sfrifle_fire.ogg'
+	can_suppress = TRUE
+	suppressor_x_offset = 4
+	suppressed_sound = 'sound/weapons/gun/general/heavy_shot_suppressed.ogg'
+	fire_select_modes = list(SELECT_SEMI_AUTOMATIC, SELECT_BURST_SHOT)
+	burst_size = 2
+	fire_delay = 3
+	pin = /obj/item/firing_pin/implant/pindicate
+	company_flag = COMPANY_REMOVED
+
+/obj/item/gun/ballistic/automatic/assault_ops_rifle/Initialize(mapload)
+	. = ..()
+
+	AddComponent(/datum/component/scope, range_modifier = 1.5)
+
+	pixel_x = -8
```

Comment:
why can't this be on the gun itself

Reply:
was done to see if it'd fix the armament vendor preview but it didn't, ill move it back